### PR TITLE
 Remove TaskManager and rely on Promises instead

### PR DESCRIPTION
**Pull request by Geod24**
There is no need for such a construct in Javascript,
as the runtime provides such a mechanism, unlike in D.

The `TaskManager` seems to originate from Agora's similar concept,
which is simply a mean to abstract away the underlying event loop,
be it Vibe.d or Agora.

In Javascript, asynchronicity is part of the language,
e.g. `window.setInterval` / `window.setTimeout`,
hence we don't swap implementation like Agora would.
Instead of re-building an event-loop within JS,
just use the native constructs.

### CODE REVIEW

File: src/Stoa.ts [hunk 1]
```diff
@@ -448,11 +445,9 @@ class Stoa extends WebService
...
                     {
                         // Recovery is required for blocks that are not received.
-                        let success: boolean = await this.recoverBlock(block, height, expected_height);
-                        if (!success)
-                        {
-                            this.pool.unshift(stored_data);
-                        }
+                        let success: boolean = false;
+                        while (!success)
+                            success = await this.recoverBlock(block, height, expected_height);
```
Comment (Geod24) [inline src/Stoa.ts:450]: I'm not 100% sure of this retry logic though. It matches the previous retry logic, but we should re-evaluate this in the future.

File: src/Stoa.ts [hunk 2]
```diff
@@ -32,14 +31,14 @@ class Stoa extends WebService
...
     /**
-     * The temporary storage of the data received from Agora
-     */
-    protected pool: Array<IPooledData>;
-
-    /**
-     * The task manager that periodically executes registered a task function
+     * Chain of pending store operations
+     *
+     * To ensure switft response time to Agora when our handlers are called,
```
Comment (AndrejMitrovic) [inline src/Stoa.ts]:
```suggestion
     * To ensure swift response time to Agora when our handlers are called,
```

**Review by AndrejMitrovic — CHANGES_REQUESTED** [review]
Makes sense to me. Just a typo to fix.

**Comment by Geod24** [comment]
Fixed typo, and extended commit description / title.

**Pull request merged**